pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -383,27 +411,36 @@ func (r *ControlPlaneReconciler) renderCharts() error {
 }
 
 func (r *ControlPlaneReconciler) PostStatus() error {
-	return r.Client.Status().Update(context.TODO(), r.Instance)
+	instance := &v1.ServiceMeshControlPlane{}
+	if updateErr := r.Client.Get(context.TODO(), client.ObjectKey{Name: r.Instance.Name, Namespace: r.Instance.Namespace}, instance); updateErr == nil {
```

Comment:
thanks for adding this, I was still seeing occasional stale object update errors

Reply:
Note that this still doesn't completely fix this. As I've mentioned before, the `get()` reads the object from the cache, which often still has the stale version of the object. The `update()` will therefore fail again. 

This is exactly what I've just seen. 

But even though this change doesn't fix the problem, it doesn't make it worse. We'll have to fix this everywhere anyway.